### PR TITLE
Split cells when neighbouring inaccessible cells

### LINKED ISSUE #264: Path taking wrong side of inaccessible cells
Further splitting near inaccessible cells is required during vessel performance (@foxm1's example)

### PR DESCRIPTION
**Pull request by gecoombs**
# PolarRoute Pull Request

Date: 26/03 
Version Number: 0.4.0
 
## Description of change
Adds a method to the VesselPerformanceModeller that will split any cells neighbouring inaccessible cells (including land cells) to match the size of the inaccessible cell.

## Fixes #264 

# Testing
Files Changed:
```
polar_route/vessel_performance/vessel_performance_modeller.py
```

- [x] My changes require one or more test files to be updated for all regression tests to pass.   

Failing tests:
[vessel_test_results.txt](https://github.com/antarctica/PolarRoute/files/14757332/vessel_test_results.txt)

Updated test results:
[vessel_tests_updated.txt](https://github.com/antarctica/PolarRoute/files/14759377/vessel_tests_updated.txt)

Unit test results:
[vessel_unit_test_dump.txt](https://github.com/antarctica/PolarRoute/files/14759413/vessel_unit_test_dump.txt)

# Checklist

- [x] My code follows [pep8](https://peps.python.org/pep-0008/) style guidelines.  
- [x] I have commented my code, particularly in hard-to-understand areas.  
- [x] I have updated the documentation of the codebase where required.  
- [x] My changes generate no new warnings.   
- [x] My PR has been made to the `0.4.x` branch (**DO NOT SUBMIT A PR TO MAIN**)

### CODE REVIEW

**Comment by gecoombs** [comment]
This PR causes various regression tests to fail as the additional splitting modifies the vessel performance mesh and its neighbour graph. I have not updated the test files on this branch yet as the final changes @Ulvetanna needs to make to resolve [the issue with splitting around waypoints](https://github.com/antarctica/MeshiPhi/issues/39) will also require changing the tests and it may be easier to roll all of these changes into one big update.

**Comment by gecoombs** [comment]
Difference in the meshes for the `grf_normal` mesh from the regression test examples:

Reference mesh:
Screenshot from 2024-03-26 11-20-35

New mesh:
Screenshot from 2024-03-26 11-20-45

**Comment by gecoombs** [comment]
I have now modified the vessel performance regression test files to match this version of the code as any further changes related to waypoint splitting will only change the route planning test results.

**Review by SamuelHall700 — APPROVED** [review]
Looks good!

File: polar_route/vessel_performance/vessel_performance_modeller.py [hunk 1]
```diff
@@ -76,4 +81,45 @@ def filter_nans(self):
...
+
+        """
+        for in_node in inaccessible_nodes:
+            in_cb = self.env_mesh.get_cellbox(in_node)
+            neighbour_nodes = self.get_all_neighbours(in_node)
+            neighbours = [self.env_mesh.get_cellbox(nn) for nn in neighbour_nodes]
+            # Only interested in splitting accessible neighbours
+            acc_neighbours = [n for n in neighbours if not n.agg_data['inaccessible']]
+            # Assume aspect ratios match and compare widths
+            while any(neighbour.boundary.get_width() > in_cb.boundary.get_width() for neighbour in acc_neighbours):
```
Comment (Ulvetanna) [inline polar_route/vessel_performance/vessel_performance_modeller.py:101]: Could we have this more generalisable for the get_width and get_height. I think in our splitting at the moment it is ok, but may create an issues later :)

**Review by Ulvetanna — CHANGES_REQUESTED** [review]
Add in the get_height(), don't only consider on the get_width alone 👍

**Review by Ulvetanna — APPROVED** [review]
Thanks for adding the addition for cell height ! All good to go

**Pull request merged**